From 42ecc15386869684cf29881a3a6941bafaa3bf69 Mon Sep 17 00:00:00 2001
From: Axel Gembe <ago@bastart.eu.org>
Date: Wed, 14 May 2008 00:25:28 +0200
Subject: [PATCH] bcm963xx: remove obsolete timer code

This removes some code that has been deprecated in kernels >= 2.6.24.

Signed-off-by: Axel Gembe <ago@bastart.eu.org>
---
 arch/mips/bcm963xx/setup.c |    2 --
 arch/mips/bcm963xx/time.c  |   29 -----------------------------
 2 files changed, 0 insertions(+), 31 deletions(-)

Index: linux-2.6.25.4/arch/mips/bcm963xx/setup.c
===================================================================
--- linux-2.6.25.4.orig/arch/mips/bcm963xx/setup.c
+++ linux-2.6.25.4/arch/mips/bcm963xx/setup.c
@@ -465,8 +465,6 @@ void __init plat_mem_setup(void)
 	_machine_halt = brcm_machine_halt;
 	pm_power_off = brcm_machine_halt;
 
-	//board_time_init = brcm_time_init;
-
     	/* mpi initialization */
     	mpi_init();
 }
Index: linux-2.6.25.4/arch/mips/bcm963xx/time.c
===================================================================
--- linux-2.6.25.4.orig/arch/mips/bcm963xx/time.c
+++ linux-2.6.25.4/arch/mips/bcm963xx/time.c
@@ -71,26 +71,6 @@ static inline unsigned long __init cal_r
 	return (mips_hpt_frequency / HZ);
 }
 
-
-/*
- * There are a lot of conceptually broken versions of the MIPS timer interrupt
- * handler floating around.  This one is rather different, but the algorithm
- * is provably more robust.
- */
-#if 0
-irqreturn_t brcm_timer_interrupt(struct pt_regs *regs)
-{
-	int irq = MIPS_TIMER_INT;
-
-	irq_enter();
-	kstat_this_cpu.irqs[irq]++;
-
-	timer_interrupt(irq, regs);
-	irq_exit();
-	return IRQ_HANDLED;
-}
-#endif
-
 void __init plat_time_init(void)
 {
 	unsigned int est_freq, flags;
@@ -107,12 +87,3 @@ void __init plat_time_init(void)
 		   (est_freq % 1000000) * 100 / 1000000);
 	local_irq_restore(flags);
 }
-
-#if 0
-void __init plat_timer_setup(struct irqaction *irq)
-{
-	r4k_cur = (read_c0_count() + r4k_offset);
-	write_c0_compare(r4k_cur);
-	set_c0_status(IE_IRQ5);
-}
-#endif
